modules/tensor_mechanics/src/postprocessors/NormalBoundaryDisplacement.C

```diff
@@ -0,0 +1,140 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "NormalBoundaryDisplacement.h"
+
+#include "libmesh/quadrature.h"
+
+registerMooseObject("TensorMechanicsApp", NormalBoundaryDisplacement);
+
+InputParameters
+NormalBoundaryDisplacement::validParams()
+{
+  InputParameters params = SidePostprocessor::validParams();
+  params.addRequiredCoupledVar("displacements", "The names of the displacement variables");
+  MooseEnum type_options("average=0 absolute_average=1 max=2 absolute_max=3", "average");
+  params.addParam<MooseEnum>("value_type", type_options, "Type of extreme value to return.");
+  params.addParam<bool>(
+      "normalize",
+      true,
+      "Computes a relative measure of normal displacement by dividing the results by sqrt(area)");
+  params.addClassDescription(
+      "This postprocessor computes the normal displacement on a given set of boundaries.");
+  return params;
+}
+
+NormalBoundaryDisplacement::NormalBoundaryDisplacement(const InputParameters & parameters)
+  : SidePostprocessor(parameters),
+    _value_type(getParam<MooseEnum>("value_type")),
+    _normalize(getParam<bool>("normalize")),
+    _ncomp(coupledComponents("displacements"))
+{
+  if (_ncomp != _mesh.dimension())
+    paramError("displacements", "Number of entries must match the mesh dimension.");
+
```

Comment:
IS this PP only supposed to work with a displaced mesh? If so please include that check here

Reply:
It seems like it should work either with the displaced or undisplaced mesh.  It should probably use the displaced mesh by default.

Reply:
It calls mesh dimension....does it really matter?

Reply:
a param error on a parameter that isnt passed isnt pretty failing with MOOSE I think

Reply:
well displacement is required so nevermind